File: iSeconds.Domain/ViewModel/OptionsList.cs
using System;
using System.Windows.Input;
using System.Collections.Generic;

namespace iSeconds.Domain
{
	public class OptionsList
	{
		public class OptionsEntry
      {
         public OptionsEntry(string name, Action callback)
         {
            this.Name = name;
            this.Callback = callback;
         }

         public string Name { get; set; }
         public Action Callback { get; set; }
      }

      private List<OptionsEntry> entries = new List<OptionsEntry>();

      public List<OptionsEntry> OptionsEntries
      {
         get { return entries; }
      }

      public ICommand EntryClicked
      {
         get
         {
            return new Command((object arg) =>
					{
						int index = (int) arg;
						entries[index].Callback.Invoke();
					});
         }
      }

		public void AddEntry(OptionsEntry entry)
		{
			entries.Add(entry);
		}

		public string[] ListNames()
		{
			string[] names= new string[OptionsEntries.Count];
			for (int i = 0; i < OptionsEntries.Count; i++)
				names[i] = OptionsEntries[i].Name;

			return names;
		}
	}
}

File: jYester/src/iSeconds/Droid/OptionsList.java
package iSeconds.Droid;

import java.util.ArrayList;
import java.util.List;

public class OptionsList 
{
	private IOptionCallback callback = null;
	private List<String> entries = new ArrayList<String>();
	
	public OptionsList(IOptionCallback callback) {
		this.callback = callback;
	}
	
	public void add(String name) {
		entries.add(name);
	}

	public CharSequence[] ToItems() {
		CharSequence[] items = new CharSequence[entries.size()];
		
		int index= 0;
		for (String entry: entries) {
			items[index] = entry;
			index++;
		}
					
		return items;
	}

	public void invoke(int which) {
		callback.invoke(which);
	}
}
